experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/types.ts

```diff
@@ -188,9 +192,10 @@ export type SharedTreeMap<TSchema extends MapSchema> = Map<string, ProxyNode<TSc
  */
 export type ProxyField<
 	TSchema extends FieldSchema,
+	API extends "javaScript" | "sharedTree" = "sharedTree",
 	// If "notEmpty", then optional fields will unbox to their content (not their content | undefined)
 	Emptiness extends "maybeEmpty" | "notEmpty" = "maybeEmpty",
-> = ProxyFieldInner<TSchema["kind"], TSchema["allowedTypes"], Emptiness>;
+> = ProxyFieldInner<TSchema["kind"], TSchema["allowedTypes"], API, Emptiness>;
```

Comment:
FYI - Is 'Emptiness' something we want to keep?  While debugging type recursion, I noticed it wasn't consistently plumbed through.

Reply:
Hmm. It is used specifically when iterating maps with "optional" (not "sequence") values. Since we only iterate the fields that have something in them, they don't need to be optional, like they would be if you simply unbox a random field/value of a map. When we write map proxies, I think it's the same. Iteration is never optional, but reading values by key is. So I think it will still be a convenient way to specify that.

Alternatively, if we changed the paradigm of maps to be "they don't have infinite entries, they actually only have the entries that they have", then the field kind could be "required", and this type could go away. But that is a more invasive change into EditableTree2.